submit hybrid ISO image support

cedwards opened the pull request:
This patch adds a new file `image_hybrid.sh` that generates a hybrid ISO capable to boot from USB.

Tested on my system using:
`poudriere image -f poudriere-image.txt -h bastillebsd -j 13amd64K -n bastillebsd -o . -t hybrid`

This created a `bastillebsd.iso` which successfully booted from USB.

bdrewery committed:
Stop using logfile birthtime for durations.

Some older FreeBSD do not support passing this value over NFSv4
nor over NFSv3.  This fixes wildly wrong durations showing in
some logs and statuses.

Commit:
Add hybrid ISO image support

This makes a version of .iso that can also be written to a USB stick

Sponsored-by: Zenith Electronics, LLC
Sponsored-by: Klara Inc.

allanjude commented on src/share/poudriere/image_hybrid.sh:
```diff
@@ -0,0 +1,97 @@
+		err 1 "Name can only contain alphanumeric characters"
+		;;
+	esac
+
```
We probably need to either restrict this to i386/amd64, or maybe remove the legacy boot support and make it EFI only, so it can support ARM64 etc.

allanjude commented on src/share/poudriere/image.sh:
```diff
@@ -44,7 +45,7 @@ poudriere image [parameters] [options]
     -j jail         -- Jail
     -t type         -- Type of image can be one of
-                    -- iso, iso+mfs, iso+zmfs, usb, usb+mfs, usb+zmfs,
+                    -- hybrid, iso, iso+mfs, iso+zmfs, usb, usb+mfs, usb+zmfs,
```
I think we need to call this `hybridiso` to avoid confusion, because hybrid could mean a mix of anything and is probably not specific enough for this context.

allanjude commented on src/share/poudriere/image_hybrid.sh:
```diff
@@ -0,0 +1,97 @@
+	# Drop the PMBR, GPT, and boot code into the System Area of the ISO.
+	dd if="${OUTPUTDIR}/hybrid.img" of="${OUTPUTDIR}/${FINALIMAGE}" bs=32k count=1 conv=notrunc
+	rm -f "${OUTPUTDIR}/hybrid.img"
+}
```
When you tested, were you able to test both Legacy and UEFI boot?

bdrewery committed:
exit_handler: Alert if an unhandled error (set -e) is seen.

Issue #985

bdrewery committed:
pkg repo: Be more verbose and catch errors better.

Issue #985

bdrewery committed:
exit_handler: Show >0 exit status always

Issue #985

bdrewery committed:
Raise test system versions

bdrewery committed:
Fix test port build on cirrus-ci

bdrewery committed:
Tweak error spam on exit to avoid false-positives.

Issue #985

bdrewery committed:
append_make: Display computed path rather than real

bdrewery committed:
options: Add -o flag to specify PORT_DBDIR name to write to.

-o port_dbdir will expand to $POUDRIERE_ETC/$port_dbdir. The
-jpz flags will be used for the make.conf, poudriere.conf, and
ports tree.

Fixes #111
Fixes #583
Fixes #176

bdrewery committed:
Minor spelling fix.

bdrewery committed:
options: Clarify how to modify the options

showconfig suggests 'make config' which can be confusing.

Fixes #992
